src/averageAges.js

```diff
@@ -55,10 +68,14 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter(({ mother, sex }) =>
+    mother && ((onlyWithSon && sex === 'm') || (!onlyWithSon))
+      && people.find(mom => mother === mom.name));
```

Comment:
Now everything is written correctly, but such code is difficult to read and maintain in the future. Break it into smaller parts.